Standardise usage requirements and assessment (#7764)
github/linguist: CONTRIBUTING.md

   We prefer examples of real-world code showing common usage.
   The more representative of the structure of the language, the better.

  > [!IMPORTANT]
  > "Hello world" and other examples commonly used in learning a language will not be accepted.

3. Open a pull request, linking to a [GitHub search result][search-example] showing in-the-wild usage.
   If you are adding a sample, please state clearly the license covering the code.
   If possible, link to the original source of the sample.
   If you wrote the sample specifically for the PR and are happy for it to be included under the MIT license that covers Linguist, you can state this instead.

  > [!IMPORTANT]

>   > Pull requests will not be reviewed if the template is not used or not filled in. 

Author: > Shouldn't it be possible to configure GitHub Actions to check if a new pull-request at least _appears_ to have filled out the template (or the relevant part of it) correctly?

Yeah, that's next on my list of things to look into when I've got a bit more time. This give us a point of reference until then.

   This command will analyze the grammar and, if no problems are found, add it to the repository.
   If problems are found, please report them to the grammar maintainer as you will otherwise be unable to add it.

  > [!IMPORTANT]
  > Please only add grammars that have [one of these licenses][licenses].

3. Add samples for your language to the [samples directory][samples] in the correct subdirectory.
   We prefer examples of real-world code showing common usage.
   The more representative of the structure of the language, the better.

  > [!IMPORTANT]

>   > "Hello world" and other examples commonly used in learning a language will not be accepted.

Reviewer: ```suggestion
  > "Hello world" and other examples found in tutorials will not be accepted.
```

Once you've received consensus that the community is happy with your proposed color change, please feel free to open a PR making the change and link to the public discussion where this was agreed by the community.
If there are official branding guidelines to support the colour choice, please link to those too.

## Language extension and filename usage requirements

In order to a new language extension or filename to be accepted in Linguist, we require that there is sufficient wide-spread usage on public GitHub repositories.

> This means we do not accept PRs for very new or hobby languages and will close PRs that are clearly very new or hobby languages.

Reviewer: ```suggestion
This means we do not accept PRs for very new or hobby languages, and will close any such PRs that attempt to add them.
```

> We use GitHub's Search to assess the popularity and rely on the search query we ask you to provide in the PR template to support the claim your language usage meets our requirements; the better your search query, the more likely your PR will be accepted.

Reviewer: I think this wording would confuse users, especially first-time contributors who aren't sure what constitutes a better or worse search query. Best to keep it simple, IMHO.
